src/components/Button/Button.tsx

```diff
@@ -39,69 +37,61 @@ export const Button = ({
   label,
   loading = false,
   disabled,
-  showLabelWithLoading = false,
   ...delegated
 }: ButtonProps) => (
   <StyledButton
     $styleType={type}
     $align={align}
     $fillWidth={fillWidth}
+    $loading={loading}
     disabled={disabled || loading}
     aria-disabled={disabled || loading}
     role="button"
     {...delegated}
   >
-    {!loading && (
-      <>
-        {iconLeft && (
-          <ButtonIcon
-            name={iconLeft}
-            aria-hidden
-            size="sm"
-          />
-        )}
-
-        {label ?? children}
-
-        {iconRight && (
-          <ButtonIcon
-            name={iconRight}
-            aria-hidden
-            size="sm"
-          />
-        )}
-      </>
+    {iconLeft && (
+      <ButtonIcon
+        name={iconLeft}
+        aria-hidden
+        size="sm"
+      />
     )}
-    {loading && (
```

Comment:
This is a breaking change as it will require consumers to change where they show progress if an action requires waiting. See onboarding, for example:

<img width="575" height="88" alt="Image" src="https://github.com/user-attachments/assets/322cb1c5-096c-44d5-87bb-f01aa97524a7" />

Reply:
nevermind. The shimmer in the CSS should handle this same experience :)